tests: driver: adc: change the sampling delay

SavinayDharmappa opened the pull request:
patch fix the issue of unable to read from adc by setting
appropriate sampling delay.

Signed-off-by: Savinay Dharmappa <savinay.dharmappa@intel.com>

nashif reviewed (changes requested):
Jira:RTOS-2360, 

please remove from commit

SavinayDharmappa committed:
tests: driver: adc: change the sampling delay

patch fix the issue of unable to read from adc by setting
appropriate sampling delay.

The sampling delay should be set to resolution of adc + 2.
In adc qmsi driver the sampling delay received form application
is subtracted from resolution of adc + 2 , if the resultant value
is less than zero then zero is set as sampling delay. So in the
adc_simple application when 12 is passed as sampling delay driver
sets it as zero. So 26 is the minimum sampling delay should be
set in application for a adc with resolution of 12. The reason
to choose 30 was to maintain same value of sampling
delay between adc_simple and adc_api application.

Signed-off-by: Savinay Dharmappa <savinay.dharmappa@intel.com>

SavinayDharmappa commented:
please review the patch

nashif commented on tests/drivers/adc/adc_simple/src/main.c:
```diff
@@ -36,7 +36,7 @@
 
 static struct adc_seq_entry sample = {
-	.sampling_delay = 12,
+	.sampling_delay = 30,
```
where does 30 come from?

SavinayDharmappa replied:
The sampling delay should be set to resolution of adc + 2.  In adc qmsi driver the sampling delay received form application is subtracted from  resolution of adc + 2 , if the resultant value is less than zero then zero is set as sampling delay. So in the adc_simple application when 12 is passed as sampling delay driver sets it as zero. So 26 is the minimum sampling delay should be set in application for a adc with resolution of 12. The reason to choose 30 was to maintain same value of sampling delay between adc_simple and adc_api application.

nashif replied:
ok, you should have this explanation in the commit message. That way we would know where this number is coming from.

SavinayDharmappa replied:
ok. I will add it

youvedeep replied:
Also please add some brief explanation what "sampling_delay " signify in this github discussion.

SavinayDharmappa replied:
ok. i will add it
